SparkPost/2web2ui
# SA-1753 - (UI) Compare By - Set up infrastructure for react-query

[SA-1753](https://sparkpost.atlassian.net/browse/SA-1753)

- [x] **NOTE:** This is rebased with `FE-1208` - that ticket should be merged/deploy prior to this one

### What Changed
- Adds `react-query` and `react-query-devtools` packages
- Adds React Query devtools and query cache provider near the root of the app
- Adds new `useSparkPostQuery` custom hook and some Cypress tests to support it
- Incorporates `useSparkPostQuery` in to the usage page (`/usage`), removing Redux
- Fixes a few issues with the usage page Cypress tests that should reduce flakiness
- Fixed problem with Cypress custom `cy.stubAuth()` command in which a request and fixture were incorrectly associated

### How To Test
- Go to `/usage` and verify the page still works as expected
- Read through the docs and make sure the implementation makes sense
- Ensure `useSparkPostQuery` handles behavior currently managed by `sparkPostApiRequest`
- Is this enough for the SA Compare By epic?

### To Do
- [x] Ensure React Query dev tools do not render for production build
- [x] Add documentation, re: using `useSparkPostQuery`
- [x] Fix test failures caused by snackbars not rendering
- [x] Rebuild branch once `FE-1208` is merged
